interpreter/exec/simd.ml

```diff
@@ -186,7 +188,13 @@ struct
     type t = Rep.t
     type lane = Float.t
     let unop f x = Convert.of_shape (List.map f (Convert.to_shape x))
+    let unopi f x = Convert.of_shape (List.mapi f (Convert.to_shape x))
     let binop f x y = Convert.of_shape (List.map2 f (Convert.to_shape x) (Convert.to_shape y))
+
+    let splat x = Convert.of_shape (List.init Convert.num_lanes (fun i -> x))
+    let extract_lane i s = List.nth (Convert.to_shape s) i
+    let replace_lane v lane r = unopi (fun i x -> if i = lane then r else x) v
```

Comment:
Nit: Can we use similar naming for similar things? For example, v for SIMD value, i,j for lane indices, x,y for lane values?

I'd also suggest tweaking the argument orders of extract and replace to be consistent with each other.

Reply:
Yea good point, done!

Reply:
Btw, made them both take the imm as first argument: `extract_lane imm v`, `replace_lane imm v x`